src/aurelia-slickgrid/services/universalTranslate.service.ts

```diff
@@ -0,0 +1,39 @@
+import { TranslaterService, TranslateServiceEventName } from '@slickgrid-universal/common';
+import { inject } from 'aurelia-framework';
+import { I18N } from 'aurelia-i18n';
+
+interface Locales {
+  [locale: string]: string;
+}
+
+interface TranslateOptions {
+  loadPath: string;
+  lang: string;
+}
+
+/**
+ * This is a Translate Service Wrapper for Slickgrid-Universal,
+ * it must implement Slickgrid-Universal TranslaterService interface to work properly
+ */
+@inject(I18N)
+export class UniversalTranslateService implements TranslaterService {
+  eventName = 'onLanguageChange' as TranslateServiceEventName;
+
+  constructor(private i18n: I18N) { }
+
+  getCurrentLanguage(): string {
+    return this.i18n.getLocale();
+  }
+
+  async use(newLang: string): Promise<Locales> {
+    return this.i18n.setLocale(newLang);
+  }
+
+  setup(options: TranslateOptions) {
+    this.i18n.setup(options);
```

Comment:
translate options contains lang (guess you meant `lng`) and `loadPath` but this is a configuration of the backend and can't be passed that way. So I'm not sure whether this is a compatibility related introduction or just an oversight.

Reply:
ah yeah I actually didn't look at the full content of that file, I added that in Slickgrid-Universal long time ago and copied over that same file. I'll give another look at it later and reply back, if you think we don't need it then it could be removed. The main point is that the `UniversalTranslateService` wrapper is the most important thing so that both lib works with their own translater service and so far I have achieve that goal, I just forgot about the options part :)

Reply:
I've put a print screen of how I use this setup, I'm assuming you can replace the `TranslateOptions` with the correct I18N Type of I18N `.setup()`, I wrote this wrapper in a hurry and thought I was done when I saw it started working but obviously didn't test everything, I haven't tried this setup options yet.

Reply:
Alright then I'd give it the same type as i18n.setup has.